components/admin/lessons/AdminLessonInputs.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useState } from 'react'
+import { useAddModuleMutation } from '../../../graphql'
+import { formChange } from '../../../helpers/formChange'
+import { FormCard, MD_INPUT, Option } from '../../FormCard'
+import { AlertFillIcon, CheckCircleIcon } from '@primer/octicons-react'
+import styles from '../../../scss/adminLessonInputs.module.scss'
+import { Spinner } from 'react-bootstrap'
+import { GraphQLError } from 'graphql-request/dist/types'
+
+type Props = {
+  lesson: { title: string; id: number }
+}
+
+enum Error {
+  InvalidData = 'missing module name or description'
+}
+
+type FormOptions = (Option & { value?: string })[]
+
+const values: FormOptions = [
+  {
+    title: 'Module Name'
+  },
+  {
+    title: 'Description',
+    type: MD_INPUT
+  }
+]
+
+const AdminLessonInputs = ({ lesson }: Props) => {
+  const [formOptions, setFormOptions] = useState(values)
+  const [name, content] = formOptions
+
+  const [addModuleMutation, { data, error, loading }] = useAddModuleMutation({
+    variables: {
+      content: content.value || '',
+      lessonId: lesson.id,
+      name: name.value || ''
+    }
+  })
+
+  const [errorMsg, setErrorMsg] = useState(error?.message || '')
+
+  const handleChange = async (value: string, propertyIndex: number) => {
+    await formChange(value, propertyIndex, formOptions, setFormOptions)
+  }
+
+  const onSubmit = async () => {
+    try {
+      // Empty strings aren't handled in the resolver
+      if (!content.value || !name.value) {
+        return setErrorMsg(Error.InvalidData)
+      }
+
+      setErrorMsg('')
+      await addModuleMutation()
+    } catch (err) {
+      setErrorMsg((err as GraphQLError).message)
+    }
+  }
+
+  const QueryStateMessage = () => {
+    if (loading) {
+      return (
+        <div className={styles.loading}>
+          <Spinner animation="grow" size="sm" />
+          <span>Adding the module...</span>
+        </div>
+      )
+    }
+
+    if (errorMsg || error) {
```

Comment:
Is the `errorMsg ||` needed? Why are we doing `errorMsg || error`?

Reply:
`errorMsg` is [set](https://github.com/garageScript/c0d3-app/pull/1809/files#diff-ce1a23c3d41862ff0962cd9bad62e18be06b4c3b5d8c03fb303e6e67fb24c99aR52) as a custom error message when the inputs are empty because the resolver doesn't validate the arguments.

It also [set](https://github.com/garageScript/c0d3-app/pull/1809/files#diff-ce1a23c3d41862ff0962cd9bad62e18be06b4c3b5d8c03fb303e6e67fb24c99aR58) the query error.

I think it could be `if (errorMsg)`.